File: backend/src/routes/acompanhamentoSCMRoutes.js

const express = require('express');
const router = express.Router();
const acompanhamentoSCMController = require('../controllers/acompanhamentoSCMController');
// Novo: Listar histórico de geração de CSV SCM
router.get('/historico/csv', acompanhamentoSCMController.getSCMHistoricoCSV);

// Novo: Adicionar entrada ao histórico de geração de CSV SCM
router.post('/historico/csv', acompanhamentoSCMController.addSCMHistoricoCSV);

// Excluir entrada do histórico de geração de CSV SCM
router.delete('/historico/csv', acompanhamentoSCMController.deleteSCMHistoricoCSV);

// Buscar status de anos desligados/ocultos do SCM para um cliente
router.get('/:cnpj', acompanhamentoSCMController.getSCMStatus);

// Salvar status de anos desligados/ocultos do SCM para um cliente
router.post('/:cnpj', acompanhamentoSCMController.setSCMStatus);

module.exports = router;

File: python_backend/routes/acompanhamento_scm_routes.py
from fastapi import APIRouter, Path
from controllers import acompanhamento_scm_controller

router = APIRouter()

@router.get('/historico/csv')
async def get_historico_csv():
    return await acompanhamento_scm_controller.get_scm_historico_csv()

@router.post('/historico/csv')
async def add_historico_csv(payload: dict):
    return await acompanhamento_scm_controller.add_scm_historico_csv(payload)

@router.delete('/historico/csv')
async def delete_historico_csv(payload: dict):
    return await acompanhamento_scm_controller.delete_scm_historico_csv(payload)

@router.get('/{cnpj}')
async def get_status(cnpj: str = Path(...)):
    return await acompanhamento_scm_controller.get_scm_status(cnpj)

@router.post('/{cnpj}')
async def set_status(cnpj: str = Path(...), payload: dict = None):
  return await acompanhamento_scm_controller.set_scm_status(cnpj, payload or {})
